pydantic/networks.py

```diff
@@ -396,6 +396,10 @@ def apply_default_parts(cls, parts: 'Parts') -> 'Parts':
 
     @classmethod
     def quote(cls, string: str, safe: str = '') -> str:
+        pattern = r'^([\w]+|(%\d{2}))+$'
```

Comment:
please compile the pattern and then use the `match` on the compiled pattern.
Take a look at the same implementation in `datetime_parse.py`
https://github.com/pydantic/pydantic/blob/32ea8853450b50e5b963ce677ab1fff95bc2e56a/pydantic/datetime_parse.py#L30

Reply:
Can you add a comment explaining what this regex is doing?

I also think it could be simplified.